spark/src/main/java/org/apache/iceberg/actions/RemoveOrphanFilesAction.java

```diff
@@ -141,7 +141,7 @@ public RemoveOrphanFilesAction deleteWith(Consumer<String> newDeleteFunc) {
     Dataset<Row> validFileDF = validDataFileDF.union(validMetadataFileDF);
     Dataset<Row> actualFileDF = buildActualFileDF();
 
-    Column joinCond = validFileDF.col("file_path").equalTo(actualFileDF.col("file_path"));
+    Column joinCond = actualFileDF.col("file_path").contains(validFileDF.col("file_path"));
```

Comment:
Isn't this going to cause Spark to use a nested loop join (full join) because there is no way to partition the data for this expression?

To fix it, what about using just the file name as well? File names should be unique because we embed the write UUID, partition, and task ID. And if we add both checks, filename could be used to distribute the data without many collisions and contains could be used for final correctness.

```
Column nameEqual = filename(actualFileDF.col("file_path")).equals(filename(validFileDF.col("file_path")));
Column actualContains = actualFileDF.col("file_path").contains(validFileDF.col("file_path"));
Column joinCond = nameEqual.and(actualContains);
```

FYI @aokolnychyi.

Reply:
Ok, I see. If the file name is unique, then I think it would be fine to change to this way. Let me update the code.

Reply:
Even if it isn't unique, we don't expect many duplicates because writers will be operating in parallel. And the worst case is all files have the same name and all get joined in a task -- which is pretty much the same as using a nested loop join.